projects/fsastorefrontlib/src/cms-components/checkout/components/legal/legal-documents/legal-documents.component.spec.ts

```diff
@@ -0,0 +1,94 @@
+import { waitForAsync, ComponentFixture, TestBed } from '@angular/core/testing';
+import { FormsModule } from '@angular/forms';
+import { I18nTestingModule, OrderEntry, OccConfig } from '@spartacus/core';
+import { FSCart, FSProduct } from '../../../../../occ/occ-models/occ.models';
+import { Observable, of } from 'rxjs';
+import { FSCartService } from '../../../../../core/cart/facade/cart.service';
+import { LegalDocumentsComponent } from './legal-documents.component';
+
+const mockProduct: FSProduct = {
+  defaultCategory: {
+    code: 'insurances_auto',
+  },
+  configurable: false,
+};
+
+const mockEntries: OrderEntry[] = [
+  {
+    product: mockProduct,
+  },
+];
+
+const mockCart: FSCart = {
+  deliveryOrderGroups: [
+    {
+      quantity: 1,
+    },
+  ],
+  entries: [
+    {
```

Comment:
Tests were failing when I tried to mock documents as a part of `cart` object, because we have not extended interface for that. With this test coverage is 100% and tests are passing.